src/catch2/reporters/catch_reporter_listening.cpp

```diff
@@ -15,151 +17,157 @@ namespace Catch {
         m_listeners.push_back( std::move( listener ) );
     }
 
-    void ListeningReporter::addReporter(IStreamingReporterPtr&& reporter) {
-        assert(!m_reporter && "Listening reporter can wrap only 1 real reporter");
-        m_reporter = std::move( reporter );
-        m_preferences.shouldRedirectStdOut = m_reporter->getPreferences().shouldRedirectStdOut;
+    void ListeningReporter::addReporter( IStreamingReporterPtr&& reporter ) {
+        m_preferences.shouldRedirectStdOut |= reporter->getPreferences().shouldRedirectStdOut;
+
+        // We will need to output the captured stdout if there are reporters
+        // that do not want it captured.
+        m_haveNoncapturingListeners |= !reporter->getPreferences().shouldRedirectStdOut;
+
+        m_preferences.shouldReportAllAssertions |=
+            reporter->getPreferences().shouldReportAllAssertions;
```

Comment:
Listeners should probably participate in this too.

Or maybe they should even force this to be on? Given their even based design it seems weird that you can have a listener only receive some assertions... But obviously listeners might also care about completely different events, and then it is pointless to pass them assertions.

I think this might end up being quite surprising for the users, and we might need to consider having listeners default to `shouldReportAllAssertions == true`, but allow them to opt out...